oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -325,19 +326,30 @@ static ExternalAccountCredentials fromJson(
           /* scopes= */ null,
           /* environmentProvider= */ null);
     }
-    return new IdentityPoolCredentials(
-        transportFactory,
-        audience,
-        subjectTokenType,
-        tokenUrl,
-        new IdentityPoolCredentialSource(credentialSourceMap),
-        tokenInfoUrl,
-        serviceAccountImpersonationUrl,
-        quotaProjectId,
-        clientId,
-        clientSecret,
-        /* scopes= */ null,
-        /* environmentProvider= */ null);
+
+    // Validate that the userProject parameter, if present, is for a Workforce configuration.
+    // This doesn't validate the if the workforce audience is valid, only if it follows the
+    // workforce pattern.
+    Pattern workforceAudiencePattern =
+        Pattern.compile("^.+/locations/.+/workforcePools/.+/providers/.+$");
+    if (userProject != null && !workforceAudiencePattern.matcher(audience).matches()) {
+      throw new IOException(
+          "The workforce_pool_user_project parameter should only be provided for a Workforce Pool configuration.");
+    }
+
+    return IdentityPoolCredentials.newBuilder()
```

Comment:
Return either IndentityPoolCredential or WorkforcePoolCredential.. 

If done separately, we can explicitly implement and document the condition where user auth takes priority over service. Right now it is not easy to find out.

Reply:
See comment above.